src/Microsoft.AspNetCore.Sockets/Internal/ConnectionState.cs

```diff
@@ -75,6 +80,9 @@ public ConnectionState(Connection connection, IChannelConnection<Message> applic
 
             // REVIEW: Add a timeout so we don't wait forever
             await Task.WhenAll(applicationTask, transportTask);
```

Comment:
You don't mean move, you mean, add another set of waiting. The only downside is that we'd need to replicate a bunch of logic around null checking.